Sources/NIO/EventLoop.swift

```diff
@@ -512,16 +510,21 @@ internal final class SelectableEventLoop: EventLoop {
                 let now = DispatchTime.now()
 
                 // Make a copy of the tasks so we can execute these while not holding the lock anymore
-                while tasksCopy.count < tasksCopy.capacity, let task = scheduledTasks.peek(), task.readyIn(now) <= .nanoseconds(0) {
-                    tasksCopy.append(task.task)
-
-                    _ = scheduledTasks.pop()
+                while tasksCopy.count < tasksCopy.capacity, let task = scheduledTasks.peek() {
+                    if task.readyIn(now) <= .nanoseconds(0) {
+                        _ = scheduledTasks.pop()
+                        tasksCopy.append(task.task)
+                    } else {
+                        nextReadyTask = task
+                        break
+                    }
                 }
 
+                let cnt = tasksCopy.count
```

Comment:
Why did this move into the lock?

Reply:
@Lukasa because I am stupid... let me fix this its not needed.

Reply:
fixed